deepchem/feat/material_featurizers/composition_featurizer.py

```diff
@@ -0,0 +1,152 @@
+    return form
+
+  def parse_formula(self, formula):
+    stack = []
+    curr_str = ''
+    i = 0
+    res = defaultdict(int)
+    formula = formula.replace('-', '').replace('@', '').replace(
+        ' ', '').replace('[', '(').replace(']', ')').replace('{', '(').replace(
+            '}', ')').replace('@', '').replace('x', '').replace(' ', '')
+
+    def parse_simple_formula(x):
+      x = self.parse_fractions(x)
+      pairs = formulare.findall(x)
+      length = sum((len(p[0]) + len(p[1]) for p in pairs))
+      assert length == len(x)
+      formula_dict = defaultdict(int)
+      for el, sub in pairs:
+        formula_dict[el] += float(sub) if sub else 1
+      return formula_dict
+
+    while i < len(formula):
+      if formula[i] not in ['(', ')'] and not stack:
+        curr_str = ''
+        while i < len(formula) and formula[i] != '(':
+          curr_str += formula[i]
+          i += 1
+        fract = re.findall(r'\d*\.*\d*', curr_str)[0]
+        curr_str = curr_str[len(fract):]
+        if not len(fract):
+          fract = 1.
+        else:
+          fract = float(fract)
+        temp_res = parse_simple_formula(curr_str)
+        for k, v in temp_res.items():
+          res[k] = temp_res[k] if k not in res else res[k] + temp_res[k]
+      elif formula[i] not in [')']:
+        stack.append(formula[i])
+        i += 1
+      else:
+        i += 1
+        fract = re.findall(r'\d*\.*\d*', formula[i:])[0]
+        i = i + len(fract)
+        if not len(fract):
+          fract = 1.
+        else:
+          fract = float(fract)
+        curr_str = ''
+        while stack[-1] != '(':
+          curr_str += stack.pop()
+        stack.pop()
+        curr_str = curr_str[::-1]
+        fract1 = re.findall(r'\d*\.*\d*', curr_str)[0]
+        if not len(fract1):
+          fract *= 1.
+        else:
+          fract *= float(fract1)
+        curr_str = curr_str[len(fract1):]
+        temp_res = parse_simple_formula(curr_str)
+        for k, v in temp_res.items():
+          temp_res[k] *= fract
+        if not stack:
+          for k, v in temp_res.items():
+            res[k] = temp_res[k] if k not in res else res[k] + temp_res[k]
+        else:
+          for i, v in temp_res.items():
+            stack.append(i)
+            stack.append(v)
+    if any([e for e in res if e in ['T', 'D', 'G', 'M', 'Q']]):
+      print(formula, res)
+    sum_nums = 1. * sum(res.values())
+    for k in res:
+      res[k] = 1. * res[k] / sum_nums
+    return res
+
+  def _featurize(self, composition: PymatgenComposition) -> np.ndarray:
+    """
+    Calculate composition vector from composition.
+
+    Parameters
+    ----------
+    composition: pymatgen.Composition object
+      Composition object.
+
+    Returns
+    -------
+    feats: np.ndarray
+      Vector of fractional compositions of each element.
+    """
+    try:
+      pretty_comp = composition.reduced_formula
+      feats = self.get_fractions(self.parse_formula(pretty_comp))
+    except:
```

Comment:
We don't need to catch the exception. `MaterialCompositionFeaturizer` (parent class) catch the exception.